keeweb/lib/Migration/RegisterMimeType.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace OCA\Keeweb\Migration;
+
+use OCP\Files\IMimeTypeLoader;
+use OCP\Migration\IOutput;
+use OCP\Migration\IRepairStep;
+
+class RegisterMimeType extends MimeTypeMigration
+{
+    public function getName()
+    {
+        return 'Register MIME type for "application/x-kdbx"';
+    }
+
+    private function registerForExistingFiles()
+    {
+        $mimeTypeId = $this->mimeTypeLoader->getId('application/x-kdbx');
+        $this->mimeTypeLoader->updateFilecache('kdbx', $mimeTypeId);
+    }
+
+    private function registerForNewFiles()
+    {
+        $mapping = ['kdbx' => ['application/x-kdbx']];
+        $mappingFile = \OC::$configDir . self::CUSTOM_MIMETYPEMAPPING;
+
+        if (file_exists($mappingFile)) {
+            $existingMapping = json_decode(file_get_contents($mappingFile), true);
+            if (json_last_error() === JSON_ERROR_NONE) {
```

Comment:
```suggestion
            if (json_last_error() === JSON_ERROR_NONE && is_array($existingMapping)) {
```

otherwise array_merge logs a warning and returns null.